### ISSUE TITLE
Auto-scrape & populate DOI, filter for 'Has DOI'

### ISSUE THREAD

**Issue by SimonDedman** [jimmejardine/qiqqa-open-source#410]
Enhancement idea: since academic journals increasingly request one adds the DOI for references, I manually populate these but it's another chore. Seems like something which could be automated - typically it's on the first page (sometimes last, or start of the bibliography), almost always has "doi:" in it, with a reasonably characteristic format that's distinct from text.
Thoughts? Obviously this would be waaay down your to-do list!

And if implemented, ideally one could then filter by Has DOI to find those lacking it and run the tool on old papers.
Cheers!

**Comment by GerHobbelt** [comment]
Yup, seen DOI tags show up much more often the last couple of years myself, so had this bit on the backburner. 👍  
Plus the sensible consequence: a tool to extract from the paper OR to 'google it' when the document lacks a recognizable DOI key, to be applied to already-present-in-the-library documents (batch mode).

This will however have to wait until I either have crazy moment (a loooong one) or when I get out of the rabbit hole that is "Qiqqa Upgrade" -- which honestly is rather a write-it-all-again-with-feeling project as nothing (except the SQlite3 metadata database backend) of the core machinery that makes Qiqqa Qiqqa has withstood the test of time. What doesn't help is that I am *way to aware* I have set myself up for [Second System Syndrome (Mr. Brooks (R.I.P.) cries me an ocean)](https://en.wikipedia.org/wiki/The_Mythical_Man-Month). Alas, all I can do now is plod on.

Thank you for your remarkable patience; few people are willing to wait on a result that requires copious amounts of *hope* -- me having been glacially slow the last two years has consumed the other type of goodwill. *Anyway.*

Good point about the DOI tag. Will become part of the metadata / BibTeX.

By the way: if you find or know handy websites to get a DOI for a given document title/author (like google, but specialized), please post such links here as I've found (with my interests) that a notable percentage of (older and/or non-academic origin) documents turn out to have DOI tags but were never published including one in the text itself, so that's another bit of metadata we'll have to dig up from the Interwebz after the fact. Like having to OCR your documents. 😢

**Comment by GerHobbelt** [comment]
https://library.centre.edu/citing_sources/DOI

**Comment by Titus7777** [comment]
I find that doing a metadata search with a subset of the BibTeX fields at [Crossref.org](https://www.crossref.org/) is a great way to find missing DOIs.  https://api.crossref.org/swagger-ui/index.html
